.pre-commit-hooks.yaml

```diff
@@ -65,3 +65,27 @@
     entry: "\uFFFD"
     language: pygrep
     types: [text]
+-   id: ts-no-debugger
+    name: check for debbuger
+    description: 'A quick check for the `debugger` statement'
+    language: pygrep
+    entry: '\bdebugger\b'
+    types: [ts]
+-   id: ts-no-console
+    name: check console functions
+    description: 'A quick check for the use of `console` built-in functions'
+    language: pygrep
+    entry: '\bconsole\.'
+    types: [ts]
+-   id: ts-no-window
+    name: check window access
+    description: 'A quick check for the use of the `window` object'
+    language: pygrep
+    entry: '\bwindow\b'
```

Comment:
isn't this a builtin check in eslint? should that be used instead?

Reply:
If only that was used everywhere. I am fine with removing it.

Reply:
you can use eslint with pre-commit if that's what you're asking: https://github.com/pre-commit/mirrors-eslint

ideally these wouldn't duplicate any default rules there

Reply:
Personally I prefer the greps for (a select few) things, as they are faster to execute (especially on every commit as opposed to every push).
If there should be no duplicates of existing eslint rules, then these checks shouldn't be added here. As mentioned above, they are equivalent to using [no-restricted-globals](https://eslint.org/docs/rules/no-restricted-globals) with the keywords added in the regexps.